docs/content/en/docs/design/config.md

```diff
@@ -156,7 +156,9 @@ The remote config gets treated like a local config after substituting the path w
 
 ### Profile Activation in required configs
 
-Additionally the `activeProfiles` stanza can define the profiles to be activated in the required configs, via:
+Skaffold will try to apply `profiles` specified by the `--profile` flag, recursively to all configurations imported as dependencies. This behavior can be disabled by setting the `--apply-profiles-recursively` flag to `false`.
```

Comment:
Maybe "propagates" instead of "recursively"?
```suggestion
Skaffold recursively applies profiles specified by the `--profile` flag to all configurations imported as dependencies. This behavior can be disabled by setting the `--apply-profiles-recursively` flag to `false`.
```

What happens if a dependent config doesn't specify that profile?  Is it silently dropped?